### PR TITLE
Technician SQL functionality

### PR DESCRIPTION
**Pull request by DanaKhromenko**
In this PR:
- Added MySQL and JSTL dependencies and maven-compiler-plugin to pom.xml;
- Improved SQL tables structure;
- Added JSP files for technicians, portfolio projects and tools;
- Added controllers for technicians, portfolio projects and tools;
- Added MySQL implementations for technicians', portfolio projects' and tools' DAOs;
- Improved getConnection() method in ConnectionUtil.

There are still some TODOs in the code. They will be implemented in the next PRs.

Until @VadimKhromenko is working on the front-end part, temporary solution looks like this:
index

add-technician

all-technicians

all-tools

### CODE REVIEW

**Comment by VadimKhromenko** [comment]
@DanaKhromenko, what are you planning to implement in this PR?

**Comment by DanaKhromenko** [comment]
> @DanaKhromenko, what are you planning to implement in this PR?

@VadimKhromenko, I plan to implement all DAO methods for Technician as well.

**Comment by VadimKhromenko** [comment]
> > @DanaKhromenko, what are you planning to implement in this PR?
> 
> @VadimKhromenko, I plan to implement all DAO methods for Technician as well.

 @DanaKhromenko, got it, thanks.

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 1]
```diff
@@ -14,8 +14,12 @@
         <th>Redirect to</th>
     </tr>
     <tr><td><a href="${pageContext.request.contextPath}/technicians/">Display All Technicians</a></td></tr>
-    <tr><td><a href="${pageContext.request.contextPath}/TBA">Sign In</a></td></tr>
-    <tr><td><a href="${pageContext.request.contextPath}/TBA">Sign Up</a></td></tr>
-</table>
```
Comment (VadimKhromenko) [inline src/main/webapp/WEB-INF/views/index.jsp:19]: @DanaKhromenko, will this logic be implemented in the next PRs?

File: src/main/java/com/gmail/danadiadius/technicians/controller/portfolioProject/AddPortfolioProjectController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String shortProjectDescription = req.getParameter("short_project_description");
+        String detailedProjectDescription = req.getParameter("detailed_project_description");
+        String sourceCodeUrl = req.getParameter("source_code_url");
+        String interactiveResultUrl = req.getParameter("interactive_result_url");
+        String pictureUrl = req.getParameter("picture_url");
+        PortfolioProject portfolioProject = new PortfolioProject(name, shortProjectDescription,
+                detailedProjectDescription, sourceCodeUrl, interactiveResultUrl, pictureUrl, Collections.emptyList());
+        // TODO: add TOOLS
```
Comment (VadimKhromenko) [inline src/main/java/com/gmail/danadiadius/technicians/controller/portfolioProject/AddPortfolioProjectController.java:36]: ok

File: src/main/webapp/WEB-INF/portfolio_projects/add.jsp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    <tr>
+        <td><label for="interactive_result_url">Interactive Result URL:</label></td>
+        <td><input type="text" id="interactive_result_url" name="interactive_result_url" form="portfolio_project"></td>
+    </tr>
+    <tr>
+        <td><label for="picture_url">Picture URL:</label></td> <%-- TODO: update to picture upload --%>
+        <td><input type="text" id="picture_url" name="picture_url" form="portfolio_project"></td>
+    </tr>
+    <tr>
+        <td><%-- TODO: place for multibox for tools--%></td>
```
Comment (VadimKhromenko) [inline src/main/webapp/WEB-INF/portfolio_projects/add.jsp:39]: ok

**Review by VadimKhromenko — APPROVED** [review]

**Pull request merged**